mirai-core-api/src/commonMain/kotlin/message/data/ShortVideo.kt

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2019-2023 Mamoe Technologies and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
+ *
+ * https://github.com/mamoe/mirai/blob/dev/LICENSE
+ */
+
+package net.mamoe.mirai.message.data
+
+import kotlinx.serialization.KSerializer
+import net.mamoe.mirai.Mirai
+import net.mamoe.mirai.contact.AudioSupported
+import net.mamoe.mirai.contact.Contact
+import net.mamoe.mirai.message.MessageSerializers
+import net.mamoe.mirai.message.data.MessageChain.Companion.serializeToJsonString
+import net.mamoe.mirai.message.data.ShortVideo.Builder
+import net.mamoe.mirai.message.data.visitor.MessageVisitor
+import net.mamoe.mirai.utils.*
+
+/**
+ * 短视频消息，指的是可在聊天界面在线播放的视频消息，而非在群文件上传的视频文件.
+ *
+ * 短视频消息分为 [OnlineShortVideo] 与 [OfflineShortVideo]. 在本地上传的短视频为 [OfflineShortVideo]. 从服务器接收的短视频为 [OnlineShortVideo].
+ *
+ * 最推荐存储的方式是下载视频文件，每次都通过上传该文件获取视频消息.
+ * 在上传视频时服务器会根据缓存情况选择回复已有视频 ID 或要求客户端上传.
+ *
+ * # 获取短视频消息示例
+ *
+ * ## 上传短视频
+ * 使用 [Contact.uploadShortVideo]，将视频缩略图和视频[资源][ExternalResource] 上传以得到 [OfflineShortVideo].
+ *
+ * ## 使用 [Builder] 构建短视频
+ * [ShortVideo] 提供 [Builder] 构建方式，必须指定 [videoId], [fileName], [fileMd5], [fileSize] 和 [fileFormat] 参数.
+ * 可选指定 [Builder.thumbnailMd5] 和 [Builder.thumbnailSize]. 若不提供，可能会影响服务器判断缓存.
+ *
+ * ## 从服务器接收
+ * 通过监听消息接收的短视频消息可直接转换为 [OnlineShortVideo].
+ *
```

Comment:
```suggestion
 * ## 从服务器接收
 * 通过监听消息接收的短视频消息可直接转换为 [OnlineShortVideo].
 * ```kotlin
 * val video: OnlineShortVideo = event.message[OnlineShortVideo]
 * ```
```